src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -241,12 +245,17 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
 
         /** Avoids excess round-tripping when reading status file. */
         static class StatusCheck extends MasterToSlaveFileCallable<Integer> {
+            private final String charset;
+            StatusCheck(String charset) {
+                this.charset = charset;
+            }
             @Override
             @CheckForNull
             public Integer invoke(File f, VirtualChannel channel) throws IOException, InterruptedException {
                 if (f.exists() && f.length() > 0) {
                     try {
-                        String fileString = Files.readFirstLine(f, Charset.defaultCharset());
+                        Charset cs = transcodingCharset(charset);
```

Comment:
Well, we do expect that IBM Java on z/OS will report a value of  ibm.system.encoding with getProperty() call.  If such property doesn't exist we will get NULL returned (that means we aren't on z/OS IBM as this property is unique for z/OS Java). And in this case all code flow will be absolutely the same as on ASCII platform, with specifying Charset.defaultCharset() in readFirstLine call.

Reply:
I am not following your logic. If we are not on z/OS then `getIBMzOsEncoding` is not even called. Therefore this part of the patch is changing a `status.txt` read call on Linux from using `Charset.defaultCharset()` to using `transcodingCharset(charset)`, which could be anything—probably an ASCII superset, but not necessarily.

Reply:
I misunderstood your question then. On platforms rather than z/OS - behavior doesn't change much at all...  it just uses a same logic as for getOutput().   it will read status file with encoding specified either by defaultCharset() for the platform encoding or from sh step encoding:  parameter.  This logic seems to be more consistent if i'm not mistaken. Indeed i can leave as it was, creating an additional check for z/OS platform and transcode status file only we are running on z/OS, np here. Just let me know if such change needed.